framework/web/helpers/CHtml.php

```diff
@@ -1168,7 +1168,7 @@ public static function checkBoxList($name,$select,$data,$htmlOptions=array())
 
 		foreach($data as $value=>$labelTitle)
 		{
-			$checked=!is_array($select) && !strcmp($value,$select) || is_array($select) && in_array($value,$select);
+			$checked=!empty($select) && !is_array($select) && !strcmp($value,$select) || is_array($select) && in_array($value,$select);
```

Comment:
I'm not sure this is safe: `empty()` will also discard any falsy value like "0". Also I'm not entirely sure null is not allowed as a valid key for $data.

If you're specifically want to catch the error from strcmp() you could probably replace that comparison with one of these:
```
($select !== null && !strcmp($value,$select))
!strcmp($value,(string)$select) // this is how it is solved elsewhere in this file, see CHtml::listOptions()
```

And could you also check radioButtonList()? It has the same `!strcmp($value,$select);` and might me missing a fix also.

Reply:
I think that `strcmp($value,(string)$select)` is the way to go - it worked like that previously, but cast to string was implicit.